PORTAL/jobs/_utils.py

```diff
@@ -3686,7 +3684,6 @@ def as_jsonable(self):
 class SSHConnectionConfig(Config):
 
     FIELDS = ['user', 'host', 'port', 'agent']
-    OPTIONAL = ['agent']
```

Comment:
This used to [be retrieved directly from the config object](https://github.com/faster-cpython/tools/blob/main/PORTAL/jobs/_job.py#L364), but the config object no longer has a single worker, it has multiple workers.  In order to get to it from there, we would need to know the `workerid` to get it out of the config.  But since we already have an `SSHConfig` object at that point, it was easier to just keep the agent information along with it.

Reply:
I'm still not sure I follow.  Why must it be required?  The logic in the function you cited indicates that it can be `None`, so why must it be required in the config?  The question of exposing the agent via the worker's SSH client is somewhat orthogonal, isn't it?

Reply:
I've removed this change related to SSH agent, and just grabbing it off of the worker-in-effect's config instead (closer to what it did before).